alibuild_helpers/init.py

```diff
@@ -82,7 +90,14 @@ def doInit(setdir, configDir, pkgname, referenceSources, dist, defaults, dryRun)
       continue
     p["ver"] = p["ver"] if p["ver"] else spec.get("tag", spec["version"])
     debug("cloning %s%s for development" % (spec["package"], " version "+p["ver"] if p["ver"] else ""))
-    updateReferenceRepos(referenceSources, spec["package"], spec)
+
+    if fetchRepos:
```

Comment:
Here again I would simplify it. The action `updateReferenceRepos` IMHO should be semantically expressed only once, with a careful specification of the conditions that lead to it. See my example for `build.py`.